### PR TITLE
google-chrome: init at 45.0.2454.101-1

### PR DESCRIPTION
**Pull request by msteen**
Added msteen to the maintainers

Please let me know if I should have rebased on master instead.

### CODE REVIEW

File: pkgs/applications/networking/browsers/google-chrome/default.nix [hunk 1]
```diff
@@ -0,0 +1,156 @@
...
+    chmod +x $exe
+
+    for elf in $out/share/google/chrome/{chrome,chrome-sandbox,nacl_helper}; do
+      patchelf --set-rpath $rpath $elf
+      patchelf --set-interpreter "$(cat $NIX_CC/nix-support/dynamic-linker)" $elf
+    done
+  '';
+
+  meta = {
+    description = "Google Chrome is a freeware web browser developed by Google.";
```
Comment (jagajaga) [inline pkgs/applications/networking/browsers/google-chrome/default.nix]: Remove dot at the end and omit pkg's name in the description.

**Comment by jagajaga** [comment]
Please squash commits into one.

cc @aszlig

**Comment by spwhitt** [comment]
> trying http://dl.google.com/linux/chrome/deb/pool/main/g/google-chrome-stable/google-chrome-stable_45.0.2454.101-1_amd64.deb
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
  0     0    0     0    0     0      0      0 --:--:-- --:--:-- --:--:--     0curl: (22) The requested URL returned error: 404 Not Found
error: cannot download google-chrome-stable_45.0.2454.101-1_amd64.deb from any mirror

**Comment by msteen** [comment]
@spwhitt Apparantly the previous stable version is removed really fast from their servers. Google Chrome 46 has been released yesterday. I have just commited an update that includes using the sources of Chrome 46.

@aszlig Thanks, I will look into that!

@jagajaga I will, once I have managed to reuse the code for the sources of Chromium plugins as @aszlig suggested.

File: pkgs/applications/networking/browsers/google-chrome/default.nix [hunk 2]
```diff
@@ -0,0 +1,190 @@
...
+    for elf in $out/share/google/chrome/{chrome,chrome-sandbox,nacl_helper}; do
+      patchelf --set-rpath $rpath $elf
+      patchelf --set-interpreter "$(cat $NIX_CC/nix-support/dynamic-linker)" $elf
+    done
+  '';
+
+  meta = {
+    description = "A freeware web browser developed by Google";
+    homepage = "https://www.google.com/chrome/browser/";
+    license = licenses.unfree;
```
Comment (domenkozar) [inline pkgs/applications/networking/browsers/google-chrome/default.nix]: is it redistributable?

**Comment by msteen** [comment]
@domenkozar In its current form it should be redistributable, because Google Chrome itself is not redistributable, but it should be legal to redistribute a recipe on how to automatically download Google Chrome and repackage it. Gentoo and Arch Linux (in their AUR) are doing the same thing. Otherwise the currently Chromium build would also be illegal, since it is doing the same thing for its plugins.

**Comment by domenkozar** [comment]
OK, then we should change license to unfreeredistributable

**Comment by msteen** [comment]
@domenkozar To be clear, as I understand it is not allowed to redistribute a repackaged Google Chrome pre-compiled. Will changing the license to unfreeredistributable mean that it will be build by Hydra? If that is the case, it should remain unfree, which I was told will cause Hydra not to build it.

**Comment by domenkozar** [comment]
Understood.

**Comment by msteen** [comment]
@aszlig I have done what you suggested, it is now leveraging the update code as used by Chromium for its plugin binaries.

@jagajaga It has been merged into one commit.

**Comment by devhell** [comment]
Will this be merged at some point?

**Comment by msteen** [comment]
@devhell That is not up to me, but to make it easier to get merged, I will close this pull request and resubmit it rebased on master instead without adding myself to the list of maintainers, which might otherwise conflict, because it has already been added since.

**Pull request closed without merging**